Horizontal Layout & Storyboard Support

Sam-Spencer opened the pull request:
Added support for UICollectionViews which scroll horizontally instead of vertically. Now `TLSpringFlowLayout` will animate and properly layout both vertical and horizontal collection views.

Improved init methods to support initialization from either a Storyboard (`initWithCoder:`) or directly from code (`init`). Both methods call a common `setup` method.

Attempted to fix issue #5 by adding the `prepareForCollectionViewUpdates:` method.

Added a storyboard to the demo project.

Updated details in the Readme.md file.

Commit:
Major Update

Added support for horizontal UICollectionViews, now allows for setup from storyboard or via code, and added a storyboard example

Commit:
Readme Update

Added information to the readme

ashfurrow commented on UICollectionView-Spring-Demo/TLSpringFlowLayout.h:
```diff
@@ -8,6 +8,14 @@
 
 #import <UIKit/UIKit.h>
 
+#define kScrollResistanceFactorDefault 900.0f
```
Could we use an `extern` variable here instead of a `#define`?

Sam-Spencer replied:
@AshFurrow Well, why would `extern` be used - can't the value be changed if it's an `extern` (I'm not familiar with using `extern`)? This is the default value and it should be unchanging. Maybe a `static` variable would be better? Maybe it shouldn't be public?

ashfurrow commented on UICollectionView-Spring-Demo/TLViewController.m:
```diff
@@ -7,39 +7,38 @@
--(UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
+- (UICollectionViewCell *)collectionView:(UICollectionView *)collectionView cellForItemAtIndexPath:(NSIndexPath *)indexPath {
     UICollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:CellIdentifier forIndexPath:indexPath];
+    if (cell) return cell;
```
I'm a little confused about what's going on here – could you clarify?

Sam-Spencer replied:
I wasn't sure about that change, but my goal with that code was this:

If the cell (specified using the identifier) in the Storyboard is valid then it will be returned. Otherwise, a cell will be created and returned. During testing I noticed that warnings would sometimes print in the log stating that, "cell with ID 'CellIdentifier' could not be found" (or something to that effect). 

This piece of code was used to alleviate that issue. However, if you think it is unnecessary or incorrect - let me know and I'll fix it.

ashfurrow replied:
Yeah, it's a matter of whoever is using the library is going to have to specify a cell identifier to be set in the storyboard's prototype cell. Can you revert this method to return the way it used to?

Sam-Spencer replied:
@AshFurrow Resolved this in the most recent commit.

ashfurrow commented:
Looks great, thanks for doing this! Just a few questions I have above.

Commit:
Added Storyboard, Minor Changes

Added Storyboard files (forgot to include in last commit), updated TLSpringFlowLayout with more documentation and an iOS 7.0 check, removed unneeded demo app code.

Commit:
Sample Project Updates

Updated sample project - added icons to tab bar and app, changed tint color to match Teehan+Lax colors, fixed collection view issue (noted in Pull Request comments).

Sam-Spencer commented:
@AshFurrow I fixed the issue with the loading of the collection view's cells. I also changed the tint color of the app to match Teehan+Lax's colors (or close to it). See commit 161a427 and commit db6e96b.

ashfurrow commented:
:beers:
